packages/@aws-cdk/app-delivery/lib/pipeline-deploy-stack-action.ts

```diff
@@ -41,6 +41,29 @@ export interface PipelineDeployStackActionProps {
    * @default ``createChangeSetRunOrder + 1``
    */
   executeChangeSetRunOrder?: number;
+
+  /**
+   * The role to use when creating and executing the ChangeSet.
+   *
+   * @default a new role is created
+   */
+  role?: iam.Role;
+
+  /**
+   * The CloudFormation Capabilities enabled for the ChangeSet.
+   *
+   * @default None
```

Comment:
I guess this is a request for upstream change then.

@skinny85  ?

Reply:
The CloudFormation documentation states that `NamedIAM` implies `IAM`, and that you need not specify both.

Reply:
That future-proofing is exactly what I'm opposing to.

Reply:
@moofish32, I believe @rix0rrr means "anonymous IAM permissions" (as opposed to "named IAM permissions").

I agree wholeheartedly with Rico and Romain here. In general, we don't do these kind of future proofing in the CDK - we try to give a nicer API for the common case.

So, I would proceed as follows here:

1. Introduce a new `enum` in this module, with 3 values: `None`, `AnonymousIAM`, `NamedIAM`.
2. Make `capabilities` in `PipelineDeployStackActionProps` be of this new `enum` type (*not* an array), with default `AnonymousIAM`.
3. After this PR, open a separate PR to move this enum to the `aws-cloudformation` package, and change the API of the CFN Actions themselves to use it.

Does this make sense?

Reply:
If I can do it all in this PR should I? I haven't looked at how difficult this would be plumb.

Reply:
Probably easier to leave it out of this PR, as it's a breaking change. But knock yourself out if you want - I don't think it will be super difficult :).